lib/dust.js

```diff
@@ -504,20 +504,17 @@
 
     blocks = this.blocks;
 
-    if (!blocks) {
-      dust.log('No blocks for context `' + key + '` in template `' + this.getTemplateName() + '`', DEBUG);
-      return false;
-    }
-
-    len = blocks.length;
-    while (len--) {
-      fn = blocks[len][key];
-      if (fn) {
-        return fn;
+    if (blocks) {
+      len = blocks.length;
+      while (len--) {
+        fn = blocks[len][key];
+        if (fn) {
+          return fn;
+        }
       }
     }
 
-    dust.log('Malformed template `' + this.getTemplateName() + '` was missing one or more blocks.');
+    dust.log('No blocks for context `' + key + '` in template `' + this.getTemplateName() + '`', DEBUG);
```

Comment:
I think getBlock's `key` is a specific block name, so you can make this message a little better and say something like "Block `key` in template `templateName` was not found"